src/queries/__tests__/role.test.tsx

```diff
@@ -77,3 +78,61 @@ test('getAllByRole, queryAllByRole, findAllByRole', async () => {
     getNoInstancesFoundMessage(NO_MATCHES_TEXT)
   );
 });
+
+describe('*ByRole with a name', () => {
+  test('Find an element that has the corresponding role and a children with the name', () => {
+    const { getByRole } = render(
+      <TouchableOpacity accessibilityRole="button" testID="target-button">
```

Comment:
All of the following tests rely on verifying explicitly set `button` role on `TouchableOpacity`. It would be good to add some tests that:
1. Test of implicitly set roles by RN (I think it should do such thing, but I'm not sure it does)
2. Test various other roles beside buttons for more real life coverage.

Reply:
> Test of implicitly set roles by RN (I think it should do such thing, but I'm not sure it does)

Surprise surprise, it not something that works^^' Event `getByRole('image')`  for `render(<Image />)` doesn't work, you need to set `accessibilityRole`. That's something that we definitely should support, but I'd say it's not tied to this PR (since I don't change the role matching). I created the issue #1130 to track that.

Reply:
How come RN not implement this by default 🤦🏻‍♂️

Reply:
So the actual screen reader should know the a11y role of `Image`, but JS is not informed.

I wonder about things like `TouchableOpacity` which afaik are implemented as composite components that renders host `View`, so they probably should have a `button` role set in JS, because it's the JS part which knows that `TouchableOpacity`'s host `View` is meant to be a button.

Reply:
> So the actual screen reader should know the a11y role of `Image`, but JS is not informed.

Yes exactly. So for `TouchableOpacity`, I'm not sure what a screen-reader would read (`button` might be the default, or not), but that's what we should rely upon for our implementation.